Source/SwiftLintFramework/Rules/Lint/DiscardedNotificationCenterObserverRule.swift

```diff
@@ -51,6 +53,11 @@ public struct DiscardedNotificationCenterObserverRule: ASTRule, ConfigurationPro
                 return []
         }
 
+        if let lastMatch = regex("append\\(").matches(in: file.contents, options: [], range: range).last?.range,
```

Comment:
This only prevents issues with `append` calls, but there are multiple ways of adding objects to a collection. Including these:

```swift
set.insert(notification)
array += [notification]
let array = [notification1, notification2, notification3]
array.insert(notification, at: 0)
```

A much easier way to prevent all of these issues is to simply check if the notification is passed to a function as a parameter. Such cases should not trigger warnings. The examples should be also extended with some additional ones like I listed above.

Reply:
I have updated to check if the notification is passed to any function.